addons/udes_stock/models/stock_location.py

```diff
@@ -0,0 +1,37 @@
+# -*- coding: utf-8 -*-
+import os.path
+from odoo import fields, models
+
+
+class StockLocation(models.Model):
+    _name = "stock.location"
+    # Add messages and abstract model to locations
+    _inherit = ["stock.location", "mail.thread", "mixin.stock.model"]
+
+    # Disable translation instead of renaming.
+    name = fields.Char(translate=False)
+    # Add tracking for archiving.
+    active = fields.Boolean(tracking=True)
+
+    def get_common_ancestor(self):
+        """
+        Returns the smallest location containing all the locations in self.
+        Locations are considered to contain themselves.
+
+        :returns:
+            The stock.location containing all the locations in self,
+            or an empty stock.location recordset if there is no such location.
+        """
+        Location = self.env["stock.location"]
```

Comment:
Added this comment as you preferred  to use self when target model is self :)
```suggestion
        
```

Reply:
Discussed in other thread, we should not use self.browse() and instead should explicitly declare the self.env["target.model"]

Reply:
As per https://github.com/unipartdigital/udes-open/pull/530#discussion_r761163318 - we prefer to use the name of the object to make it more explicit.

Understand this is a contentious issue and will take a while until we're doing it consistently.